src/model/entities/test/rundown.spec.ts

```diff
@@ -2805,12 +2804,48 @@ describe(Rundown.name, () => {
           },
           infinitePieces: new Map(),
         },
+        segments: [
+          activeSegment,
+          nextSegment,
+        ],
       } as RundownInterface)
 
       testee.setNext(activeSegment.id, otherPartInActiveSegment.id)
 
       verify(mockedNextPart.reset()).once()
     })
+
+    describe('when next part is on air', () => {
+      it('throws an active part exception', () => {
+        const activePart: Part = EntityMockFactory.createPart({ id: 'active-part-id', isOnAir: true })
+        const mockedActiveSegment: Segment = EntityMockFactory.createSegmentMock({ id: 'active-segment-id', parts: [activePart] })
+        when(mockedActiveSegment.findPart(activePart.id)).thenReturn(activePart)
+        const activeSegment: Segment = instance(mockedActiveSegment)
+        const nextPart: Part = EntityTestFactory.createPart({ id: 'next-part-id', isNext: true })
+        const nextSegment: Segment = EntityTestFactory.createSegment({ id: 'next-segment-id', isNext: true, parts: [nextPart]})
+        const rundown: Rundown = new Rundown({
+          isRundownActive: true,
+          alreadyActiveProperties: {
+            activeCursor: {
+              segment: activeSegment,
+              part: activePart,
+            },
+            nextCursor: {
+              segment: nextSegment,
+              part: nextPart,
+            },
```

Comment:
You don't need a `nextCursor` for this test so you can skip all the setup for the nextCursor.

Reply:
There is a check in rundown.setNext that throws if no nextCursor is set.